src/main/java/Storage.java

```diff
@@ -0,0 +1,168 @@
+package duke;
+
+import duke.error.DukeException;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.task.Event;
+import duke.task.Deadline;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.util.ArrayList;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    public File file;
+    public File dir;
+    public String filepath;
+
+    /**
+     * Creates a instance of Storage object
+     *
+     * @param filepath String of the file location
+     */
+    public Storage(String filepath) {
+        String[] arr = filepath.split("/");
+        dir = new File(arr[0]);
+        dir.mkdirs();
+
+        this.dir = dir;
+        this.file = new File(dir, arr[1]);
+        this.filepath = arr[1];
+    }
+
+    /**
+     * Transforms data in text file to TaskLists containing list of Tasks.
+     *
+     * @returns TaskList containing list of Tasks
+     * @throws Exception if file does not exist or loading task error
+     */
+    public ArrayList<Task> load() throws Exception {
+        try {
+            if (file.exists()) {
+                return loadTasks();
+            } else {
+                throw new DukeException("New Path: File does not exist");
+            }
+        } catch (Exception e) {
+            throw new DukeException(e.getMessage());
+        }
+    }
+
+    /**
+     * Helper Function to load() method.
+     *
+     * @return list of tasks from text file
+     * @throws IOException if task type != "T" || "D" || "E"
+     * */
+    public ArrayList<Task> loadTasks() throws IOException {
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        try {
+            FileReader reader = new FileReader(file);
+            BufferedReader bufferedReader = new BufferedReader(reader);
+
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] arr = line.split(" \\| ");
+                Task task;
+
+                switch (arr[0]) {
+                case "T": //todo
+                    task = new Todo(arr[2]);
```

Comment:
Hi, you may want to consider extracting the checking of task type out, so that the switch block doesn't have to be repeated in many different classes